### ISSUE TITLE
Documentation Ambiguity 

### ISSUE THREAD

**Issue by davidbren** [explosion/spaCy#4598]
I have gone through the docs several times but I can't understand what spaCy is doing when creating a model from scratch ? 

So I have my custom own training data & I am building a blank english model from this using the NER pipeline only.  
What type of neural network is used in this scenario ? Is it as suggested in this article ?

https://medium.com/@b.terryjack/nlp-pretrained-named-entity-recognition-7caa5cd28d7b
 
If so how are the word embedding's being produced ?
Could really do with help on this please & thanks for your great product (grovel, grovel :)) 

## Your Environment
* Operating System: 
* Python Version Used: 3.7 
* spaCy Version Used: 2.1.8
* Environment Information: RHEL 7

**Comment by adrianeboyd** [comment]
I think the best place to start is this section of the docs, especially the included video:

https://spacy.io/models#architecture

**Comment by davidbren** [comment]
Thanks for the reply, but I think this looks at the method they used to make the default models
https://spacy.io/models

In my case, I am creating a model from scratch with my NER data so I am not using the pre-built models..
When Matthew talks about the parser in the video, we are only using NER we are not using a dependency parser, I assume that parser = dependency parser here ?

**Comment by adrianeboyd** [comment]
The default models not trained any differently than you would train a custom model with your own data. We use the exact same `spacy train` script that you can use to train your own model.

The NER model uses the same general transition-based parsing framework as the dependency parser, so there is a lot of overlap in the code and features for those models.

**Comment by davidbren** [comment]
Thanks for the above comment Adriane it helps.

From the link you sent earlier https://spacy.io/models#architecture
"We first predict context-sensitive vectors for each word, this convolutional layer is shared between the tagger, parser and NER"  …

"To boost the representation, the tagger actually predicts a “super tag” .. by adding a soft max layer onto the convolutional layer..." 

So it adds a SoftMax layer onto the convolutional layer, should I think of this as, the CNN learns the embeddings & the SoftMax to classify its an entity or not ?

Is there a separate pooling layer in the CNN ?

The vectors on my tokens are blank I assumed this would be where I would see the weights which inform the embeddings?
PS. thanks again for all the help on this one..

**Comment by svlandeg** [comment]
Hi @davidbren. If you want to dive deeper into the implementation, I'm afraid for now the best solution is to actually go through the code.

https://github.com/explosion/spaCy/blob/master/spacy/pipeline/pipes.pyx shows the highlevel overview of the generic pipes we have in spaCy. Here you can find for instance the definition of the `EntityRecognizer` (abbreviated to `ner`).

Most of the detailed implementations for the NER and dependency parser can be found in the folder https://github.com/explosion/spaCy/tree/master/spacy/syntax, see for instance 
[nn_parser.pyx](https://github.com/explosion/spaCy/blob/master/spacy/syntax/nn_parser.pyx).

At some point we do hope to provide more technical documentation for these implementations, though I do believe that the current documentation is at least sufficient to get you started with your use-case.

As we like to focus this tracker on bugs and feature requests specifically, I am closing this issue for now. Please feel free to open a new issue if you run into specific technical issues during implementation.
